bookkeeper-server/src/main/java/org/apache/bookkeeper/http/BKHttpServiceProvider.java

```diff
@@ -64,6 +68,9 @@ private BKHttpServiceProvider(BookieServer bookieServer,
         ClientConfiguration clientConfiguration = new ClientConfiguration(serverConf)
           .setZkServers(serverConf.getZkServers());
         this.bka = new BookKeeperAdmin(clientConfiguration);
+
+        this.executor = Executors.newSingleThreadExecutor(
+          new ThreadFactoryBuilder().setNameFormat("BKHttpServiceThread").build());
```

Comment:
thread factory should only create daemon threads, otherwise the thread may hold the process open during shutdown. You should also shutdown the executor if possible.